src/Symfony/Component/Notifier/Bridge/Smsbox/SmsboxOptions.php

```diff
@@ -0,0 +1,380 @@
+
+        return $this;
+    }
+
+    public function sender(string $sender)
+    {
+        $this->options['sender'] = $sender;
+
+        return $this;
+    }
+
+    public function toArray(): array
+    {
+        return $this->options;
+    }
+
+    public static function validMode(string $mode): string
+    {
+        $supportedModes = [
+            self::MESSAGE_MODE_STANDARD,
+            self::MESSAGE_MODE_EXPERT,
+            self::MESSAGE_MODE_RESPONSE,
+        ];
+
+        if (!\in_array($mode, $supportedModes, true)) {
+            throw new InvalidArgumentException(sprintf('The message mode "%s" is not supported; supported message modes are: "%s".', $mode, implode('", "', $supportedModes)));
+        }
+
+        return $mode;
+    }
+
+    public static function validStrategy(int $strategy): int
+    {
+        $supportedStrategies = [
+            self::MESSAGE_STRATEGY_PRIVATE,
+            self::MESSAGE_STRATEGY_NOTIFICATION,
+            self::MESSAGE_STRATEGY_NOT_MARKETING_GROUP,
+            self::MESSAGE_STRATEGY_MARKETING,
+        ];
+        if (!\in_array($strategy, $supportedStrategies, true)) {
+            throw new InvalidArgumentException(sprintf('The message strategy "%s" is not supported; supported strategies types are: "%s".', $strategy, implode('", "', $supportedStrategies)));
+        }
+
+        return $strategy;
+    }
+
+    public static function validDate(string $date): string
+    {
+        $dateTimeObj = \DateTime::createFromFormat('d/m/Y', $date);
+        $now = new \DateTime();
+        $tz = new \DateTimeZone('Europe/Paris');
+        $dateTimeObj->setTimezone($tz);
+        $now->setTimezone($tz);
+
+        if (!$dateTimeObj || $dateTimeObj->format('Y-m-d') <= (new \DateTime())->format('Y-m-d')) {
+            throw new InvalidArgumentException('The date must be in DD/MM/YYYY format and greater than the current date.');
+        }
+
+        return $date;
+    }
+
+    public static function validDateTime(\DateTime $dateTime)
+    {
+        \Locale::setDefault('fr');
+        $now = new \DateTime();
+        $tz = new \DateTimeZone('Europe/Paris');
+        $now->setTimezone($tz);
+        $dateTime->setTimezone($tz);
+
+        if ($now > $dateTime || $dateTime > $now->modify('+2 Year')) {
+            throw new InvalidArgumentException('dateTime must be greater to the actual date and limited to 2 years in the future.');
+        }
+
+        return $dateTime;
+    }
+
+    public static function validDestIso($isoCode)
+    {
+        \Locale::setDefault('fr');
+        if (2 != \strlen($isoCode) || strtoupper($isoCode) != $isoCode) {
+            throw new InvalidArgumentException('destIso must be the ISO 3166-1 alpha 2 on two uppercase characters.');
+        }
+        try {
+            Countries::getName($isoCode);
+        } catch (\Throwable) {
+            throw new InvalidArgumentException('destIso must be a ISO 3166-1 alpha 2 code.');
+        }
+
+        return $isoCode;
```

Comment:
I am unsure about this, I would only validate the string itself for syntax and let the API return a proper error

Reply:
OK, so we've opted to simply validate the format upstream and let the API do its job.